scene/2d/physics_body_2d.cpp

```diff
@@ -456,8 +456,8 @@ void RigidBody2D::_direct_state_changed(Object *p_state) {
 		sleeping = state->is_sleeping();
 		emit_signal(SceneStringNames::get_singleton()->sleeping_state_changed);
 	}
-	if (get_script_instance())
-		get_script_instance()->call("_integrate_forces", state);
+	if (has_method("_integrate_forces"))
+		call("_integrate_forces", state);
```

Comment:
Doesn't that break the possibility to override it from scripts?

Reply:
Not at all, if you look at the code of has_method and call (in core/object.cpp), you can see that it checks the script instance first to look for/call the method. So script override is still possible, and has priority over the C++ override. I've also tested it to be sure.